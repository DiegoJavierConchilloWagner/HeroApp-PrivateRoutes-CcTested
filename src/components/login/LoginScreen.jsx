import React, { useState } from 'react';
import { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';
import './style.css'

export const LoginScreen = ({ history }) => {
    const [signUpMode, setSignUpMode] = useState(false);
    const handleMode = () => setSignUpMode(!signUpMode);

    const { dispatch } = useContext(AuthContext)

    const handleLogin = () => {

        const lastPatch = localStorage.getItem('lastPath') || '/';

        dispatch({
            type: types.login,
            payload:{
                name: 'Diego Wagner',
                email: 'Diegoo.javier.wagner@gmail.com'
            }
        });
        history.replace( lastPatch );
    }

    return (
        <div className={`${signUpMode ? "container sign-up-mode" : "container"}`}>
            <div className="forms-container">
                <div className="signin-signup">
                    <form action="#" className="sign-in-form">
                        <h2 className="title">Sign in</h2>
                        <div className="input-field">
                            <i className="fas fa-user" />
                            <input type="text" placeholder="Username" />
                        </div>
                        <div className="input-field">
                            <i className="fas fa-lock" />
                            <input type="password" placeholder="Password" autoComplete="true" />
                        </div>
                        <input type="submit" defaultValue="Login" className="btn solid" onClick={ () => handleLogin() } />
                        <p className="social-text">Or Sign in with social platforms</p>
                        <div className="social-media">
                            <a href="#?" className="social-icon">
                                <i className="fab fa-facebook-f" />
                            </a>
                            <a href="#?" className="social-icon">
                                <i className="fab fa-twitter" />
                            </a>
                            <a href="#?" className="social-icon">
                                <i className="fab fa-google" />
                            </a>
                            <a href="#?" className="social-icon">
                                <i className="fab fa-linkedin-in" />
                            </a>
                        </div>
                    </form>
                    <form action="#" className="sign-up-form">
                        <h2 className="title">Sign up</h2>
                        <div className="input-field">
                            <i className="fas fa-user" />
                            <input type="text" placeholder="Username" />
                        </div>
                        <div className="input-field">
                            <i className="fas fa-envelope" />
                            <input type="email" placeholder="Email" />
                        </div>
                        <div className="input-field">
                            <i className="fas fa-lock" />
                            <input type="password" placeholder="Password" autoComplete="true"/>
                        </div>
                        <input type="submit" className="btn" defaultValue="Sign up" />
                        <p className="social-text">Or Sign up with social platforms</p>
                        <div className="social-media">
                            <a href="#?" className="social-icon">
                                <i className="fab fa-facebook-f" />
                            </a>
                            <a href="#?" className="social-icon">
                                <i className="fab fa-twitter" />
                            </a>
                            <a href="#?" className="social-icon">
                                <i className="fab fa-google" />
                            </a>
                            <a href="#?" className="social-icon">
                                <i className="fab fa-linkedin-in" />
                            </a>
                        </div>
                    </form>
                </div>
            </div>
            <div className="panels-container">
                <div className="panel left-panel">
                    <div className="content">
                        <h3>New here ?</h3>
                        <p>
                            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Debitis,
                            ex ratione. Aliquid!
                        </p>
                        <button className="btn transparent" onClick={ () => handleMode() }>
                            Sign up
                        </button>
                    </div>
                    <img src="img/log.svg" className="image" alt="" />
                </div>
                <div className="panel right-panel">
                    <div className="content">
                        <h3>One of us ?</h3>
                        <p>
                            Lorem ipsum dolor sit amet consectetur adipisicing elit. Nostrum
                            laboriosam ad deleniti.
                        </p>
                        <button className="btn transparent" onClick={ () => handleMode() }>
                            Sign in
                        </button>
                    </div>
                    <img src="img/register.svg" className="image" alt="" />
                </div>
            </div>
        </div>
    )
}