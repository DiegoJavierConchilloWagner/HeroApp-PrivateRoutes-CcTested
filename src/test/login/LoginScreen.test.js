import React from 'react';
import { mount } from 'enzyme'
import { MemoryRouter, Route } from 'react-router-dom';
import { LoginScreen } from '../../components/login/LoginScreen';
import { AuthContext } from '../../auth/AuthContext';


describe('Pruebas en LoginScreen', () => {
    
    const history = {
        replace: jest.fn()
    };
    const contextValue={
        dispatch: jest.fn(),
        user: { logged: false }
    }  

    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <LoginScreen history={ history } />
        </AuthContext.Provider>
    )

    test('Debe de mosrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    // test('Debe de realizar el dispatch y la navegaciÃ³n', () => {
    //     wrapper.find('button').prop('onClick')();

    //     // expect( contextValue.dispatch ).toHaveBeenCalledWith({type: types.login,
            // payload:{
            //     name: 'Diego Wagner',
            //     email: 'Diegoo.javier.wagner@gmail.com'
            // }})
    // });
});
