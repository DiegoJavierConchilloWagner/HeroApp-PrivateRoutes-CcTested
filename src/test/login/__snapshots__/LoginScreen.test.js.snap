// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen Debe de mosrar el componente correctamente 1`] = `
<div
  className="container"
>
  <div
    className="forms-container"
  >
    <div
      className="signin-signup"
    >
      <form
        action="#"
        className="sign-in-form"
      >
        <h2
          className="title"
        >
          Sign in
        </h2>
        <div
          className="input-field"
        >
          <i
            className="fas fa-user"
          />
          <input
            placeholder="Username"
            type="text"
          />
        </div>
        <div
          className="input-field"
        >
          <i
            className="fas fa-lock"
          />
          <input
            autoComplete="true"
            placeholder="Password"
            type="password"
          />
        </div>
        <input
          className="btn solid"
          defaultValue="Login"
          onClick={[Function]}
          type="submit"
        />
        <p
          className="social-text"
        >
          Or Sign in with social platforms
        </p>
        <div
          className="social-media"
        >
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-facebook-f"
            />
          </a>
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-twitter"
            />
          </a>
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-google"
            />
          </a>
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-linkedin-in"
            />
          </a>
        </div>
      </form>
      <form
        action="#"
        className="sign-up-form"
      >
        <h2
          className="title"
        >
          Sign up
        </h2>
        <div
          className="input-field"
        >
          <i
            className="fas fa-user"
          />
          <input
            placeholder="Username"
            type="text"
          />
        </div>
        <div
          className="input-field"
        >
          <i
            className="fas fa-envelope"
          />
          <input
            placeholder="Email"
            type="email"
          />
        </div>
        <div
          className="input-field"
        >
          <i
            className="fas fa-lock"
          />
          <input
            autoComplete="true"
            placeholder="Password"
            type="password"
          />
        </div>
        <input
          className="btn"
          defaultValue="Sign up"
          type="submit"
        />
        <p
          className="social-text"
        >
          Or Sign up with social platforms
        </p>
        <div
          className="social-media"
        >
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-facebook-f"
            />
          </a>
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-twitter"
            />
          </a>
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-google"
            />
          </a>
          <a
            className="social-icon"
            href="#?"
          >
            <i
              className="fab fa-linkedin-in"
            />
          </a>
        </div>
      </form>
    </div>
  </div>
  <div
    className="panels-container"
  >
    <div
      className="panel left-panel"
    >
      <div
        className="content"
      >
        <h3>
          New here ?
        </h3>
        <p>
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Debitis, ex ratione. Aliquid!
        </p>
        <button
          className="btn transparent"
          onClick={[Function]}
        >
          Sign up
        </button>
      </div>
      <img
        alt=""
        className="image"
        src="img/log.svg"
      />
    </div>
    <div
      className="panel right-panel"
    >
      <div
        className="content"
      >
        <h3>
          One of us ?
        </h3>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Nostrum laboriosam ad deleniti.
        </p>
        <button
          className="btn transparent"
          onClick={[Function]}
        >
          Sign in
        </button>
      </div>
      <img
        alt=""
        className="image"
        src="img/register.svg"
      />
    </div>
  </div>
</div>
`;
